### PR TITLE
Filter out candidates with the same name but different instructions, …

### PR DESCRIPTION
**Pull request by IANTHEREAL**
## Why are these changes needed?

Filter out candidates with the same name but different instructions, file IDs, and function names. 

## Related issue number

Refer https://github.com/microsoft/autogen/issues/892#

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/agentchat/contrib/gpt_assistant_agent.py [hunk 1]
```diff
@@ -368,3 +373,48 @@ def delete_assistant(self):
...
+            tool.get("function", {}).get("name")
+            for tool in tools
+            if tool.get("type") not in ["code_interpreter", "retrieval"]
+        )
+        required_file_ids = set(file_ids)  # Convert file_ids to a set for unordered comparison
+
+        for assistant in candidate_assistants:
+            # Check if instructions are similar
+            if instructions and instructions != getattr(assistant, "instructions", None):
+                print("instructions not match")
```
Comment (gagb) [inline autogen/agentchat/contrib/gpt_assistant_agent.py:397]: Use logging instead of printing

Comment (IANTHEREAL) [reply autogen/agentchat/contrib/gpt_assistant_agent.py]: my bad, I forgot to polish

**Comment by IANTHEREAL** [comment]
@gagb @sonichi PTAL

**Comment by gagb** [comment]
> @gagb @sonichi PTAL

See this PR:
https://github.com/IANTHEREAL/autogen/pull/2

Otherwise looks good

**Review by gagb — APPROVED** [review]

**Pull request merged**